metric_namer_test.go

```diff
@@ -1110,16 +1110,14 @@ func TestMetricNamer_Build(t *testing.T) {
 					UTF8Allowed: tt.namer.UTF8Allowed,
 				}
 				gotUnitName := unitNamer.Build(tt.metric.Unit)
-				if tt.expectedUnitName != gotUnitName {
-					t.Errorf("unitNamer.Build(%q), got %q, want %q", tt.metric.Unit, gotUnitName, tt.expectedUnitName)
+				if gotUnitName != tt.wantUnitName {
+					t.Errorf("UnitNamer.Build(%q) = %q, want %q", tt.metric.Unit, gotUnitName, tt.wantUnitName)
 				}
 
 				// Verify correlation: if UnitNamer produces a non-empty unit name,
 				// it should be contained in the metric name when WithMetricSuffixes=true
-				if gotUnitName != "" && tt.namer.WithMetricSuffixes {
-					if !strings.Contains(gotMetricName, gotUnitName) {
-						t.Errorf("Metric name '%q' should contain unit name '%s' when WithMetricSuffixes=true", gotMetricName, gotUnitName)
-					}
+				if tt.namer.WithMetricSuffixes && !strings.Contains(gotMetricName, gotUnitName) {
+					t.Errorf("Metric name %q should contain unit name %q when WithMetricSuffixes=true", gotMetricName, gotUnitName)
```

Comment:
This seems to be testing the test case which I find strange :sweat_smile: 
This would never fail unless someone creates an invalid test case, right?